cloudify_agent/api/pm/base.py

```diff
@@ -300,6 +329,37 @@ def validate_optional(self):
         self._validate_autoscale()
         self._validate_host()
 
+    def _get_broker_port(self):
+        """
+        Determines the broker port if it has not been provided. Only intended
+        to be called before self.broker_port has been set and after
+        self.broker_ssl_cert has been set.
+        """
+        if self.broker_ssl_cert == '':
```

Comment:
In general, anywhere we define the SSL or non-SSL ports, if a constants/defaults file exists, we can set them there, just like in the example above.